subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarCopyActionSpec.groovy

```diff
@@ -118,11 +118,7 @@ public class TarCopyActionSpec extends Specification {
         action.execute(new CopyActionProcessingStream() {
             public void process(CopyActionProcessingStreamAction action) {
                 for (FileCopyDetailsInternal f : files) {
-                    if (f.isDirectory()) {
-                        action.processFile(f);
-                    } else {
-                        action.processFile(f);
-                    }
+                    action.processFile(f);
```

Comment:
a cursory inspection seems to indicate this is the correct fix but there is a good chance I missed the logic here and really the the first if statement should be changed.